torchrl/objectives/value/functional.py

```diff
@@ -151,19 +214,31 @@ def vec_generalized_advantage_estimate(
     *batch_size, time_steps, lastdim = not_done.shape
 
     value = gamma * lmbda
```

Comment:
If gamma and lambda are tensors but scalar we should use the fast version

Reply:
I've used
```python
def _to_scalar_if_unique(tensor: Union[float, torch.Tensor]):
    """Return a single scalar if tensor consists of a single unique element, else the source tensor is returned."""
    if isinstance(tensor, torch.Tensor):
        tensor_unique = tensor.unique()
        if tensor_unique.numel() == 1:
            return tensor_unique.item()

    return tensor
```

to catch the case when gamma/lmbda have multiple identical elements, but I'm unsure if it is worth the effort. `.unique()` runtime costs are not high compared to the other code, but if improves the runtime of a case which never occurs in reality, I should probably remove it.

What is your opinion on this?

Reply:
We'll run into issues if gamma carries a gradient. Why not just `if isinstance(gamma, number) or gamma.numel()<=1` (or numpy equivalent)?

Reply:
Oh I see what you did here sorry. I do t thing we should cover this use case. If users provide a tensor full of gammas with the same value they may know what they're doing or be totally off the road, but in either case we should not do an expensive and dangerous operation like unique (same thing as my comment above: if gamma carries a gradient this will break things)

Reply:
My point was merely that your check as it was would treat `gamma = torch.tensor(0.99)` as a fully shaped gamma of the same size of reward and done. If gamma is a tensor but a scalar (it has 1 element or less) it should not be treated as a fully shaped tensor.

Reply:
Done.